src/Build/Rebuilder.hs

```diff
@@ -31,10 +31,10 @@ type MakeInfo k = (Map k Time, Time)
 modTimeRebuilder :: Ord k => Rebuilder Applicative (MakeInfo k) k v
 modTimeRebuilder key value task fetch = do
     (modTime, now) <- get
-    let dirty = case Map.lookup key modTime of
-            Nothing -> True
-            time -> any (\d -> Map.lookup d modTime > time) (dependencies task)
-    if not dirty
+    let upToDate = case Map.lookup key modTime of
+            Nothing -> False
+            time -> all (\d -> Map.lookup d modTime < time) (dependencies task)
```

Comment:
You inverted the logic, but replaced `>` with `<`, not `<=`. Does the Eq case matter? It used to before we incremented on every step, but maybe we now know each time is distinct.

Reply:
We assume that each modification time is distinct. The assumption is not actually written down, I'll need to add a comment.